### PR TITLE
feat(video-conferencing-service): SFO-48 Project Setup, sonar fixes and file indentation

### PR DESCRIPTION
**Pull request by bhavyasf**
## Description

This pull request includes project setup for video conferencing service including dummy API response, sonar fixes, and file indentation.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [x] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by samarpan-b** [comment]
Update description with details of the PR. Remove relay from project

File: services/video-conferencing-service/package.json [hunk 1]
```diff
@@ -1,19 +1,19 @@
 {
-  "name": "@sourcefuse-service-catalog/video-conferencing-service",
-  "version": "0.0.0",
-  "description": "services",
+  "name": "videochat-service",
```
Comment (samarpan-b) [inline services/video-conferencing-service/package.json:2]: revert this

File: services/video-conferencing-service/package.json [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+  "version": "1.0.0",
```
Comment (samarpan-b) [inline services/video-conferencing-service/package.json:3]: revert this

File: services/video-conferencing-service/package.json [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
+  "description": "Microservice providing videochat",
   "keywords": [
-    "loopback-extension",
+    "loopback-application",
```
Comment (samarpan-b) [inline services/video-conferencing-service/package.json:6]: revert this

File: services/video-conferencing-service/package.json [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
     "loopback"
   ],
-  "main": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "main": "index.js",
```
Comment (samarpan-b) [inline services/video-conferencing-service/package.json:9]: revert this

File: services/video-conferencing-service/package.json [hunk 5]
```diff
@@ -25,38 +25,65 @@
     "test": "lb-mocha --allow-console-logs \"dist/__tests__\"",
     "posttest": "npm run lint",
     "test:dev": "lb-mocha --allow-console-logs dist/__tests__/**/*.js && npm run posttest",
-    "clean": "lb-clean dist *.tsbuildinfo .eslintcache"
+    "docker:build": "docker build -t videochat-service .",
```
Comment (samarpan-b) [inline services/video-conferencing-service/package.json:28]: revert this

File: services/video-conferencing-service/package.json [hunk 6]
```diff
@@ -1,19 +1,19 @@
...
-  "main": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "main": "index.js",
   "engines": {
     "node": ">=10"
   },
   "scripts": {
     "build": "lb-tsc",
     "build:watch": "lb-tsc --watch",
+    "clean": "lb-clean dist *.tsbuildinfo",
```
Comment (samarpan-b) [inline services/video-conferencing-service/package.json:16]: revert this

File: services/video-conferencing-service/index.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+const application = require('./dist');
```
Comment (samarpan-b) [inline services/video-conferencing-service/index.js:1]: revert this file

File: services/video-conferencing-service/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,28 @@
+# Check out https://hub.docker.com/_/node to select a new base image
```
Comment (samarpan-b) [inline services/video-conferencing-service/Dockerfile:1]: remove docker

File: services/video-conferencing-service/src/enums/index copy.ts [hunk 9]
```diff
@@ -0,0 +1,2 @@
+export * from './status-codes.enum';
```
Comment (samarpan-b) [inline services/video-conferencing-service/src/enums/index copy.ts:1]: incorrect file name

File: services/video-conferencing-service/src/datasources/videochat-db.datasource.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import {inject, lifeCycleObserver, ValueOrPromise} from '@loopback/core';
+import {juggler} from '@loopback/repository';
+import config from './videochat-db.datasource.config.json';
+
+@lifeCycleObserver('datasource')
+export class VideochatDbDataSource extends juggler.DataSource {
```
Comment (samarpan-b) [inline services/video-conferencing-service/src/datasources/videochat-db.datasource.ts:6]: do it similar to how notif service is done

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+    },
+  })
+  async getMeetingLink(
+    @requestBody()
+    meetingOptions: MeetingOptions,
+  ): Promise<string> {
+    return 'meetingLink';
+  }
+
+  @authorize(['*'])
```
Comment (samarpan-b) [inline services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:40]: use permissions

Comment (samarpan-b) [inline services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:40]: use authenticate bearer

File: services/video-conferencing-service/package.json (hunk 1 above)
Comment (vineet-suri) [reply services/video-conferencing-service/package.json:2]: done

File: services/video-conferencing-service/package.json (hunk 2 above)
Comment (vineet-suri) [reply services/video-conferencing-service/package.json:3]: done

File: services/video-conferencing-service/package.json (hunk 3 above)
Comment (vineet-suri) [reply services/video-conferencing-service/package.json:6]: done

File: services/video-conferencing-service/package.json (hunk 4 above)
Comment (vineet-suri) [reply services/video-conferencing-service/package.json:9]: done

File: services/video-conferencing-service/package.json (hunk 6 above)
Comment (vineet-suri) [reply services/video-conferencing-service/package.json:16]: done

File: services/video-conferencing-service/Dockerfile (hunk 8 above)
Comment (vineet-suri) [reply services/video-conferencing-service/Dockerfile:1]: done

File: services/video-conferencing-service/index.js (hunk 7 above)
Comment (vineet-suri) [reply services/video-conferencing-service/index.js:1]: done

File: services/video-conferencing-service/package.json (hunk 5 above)
Comment (vineet-suri) [reply services/video-conferencing-service/package.json:28]: done

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts (hunk 11 above)
Comment (vineet-suri) [reply services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:40]: done

Comment (vineet-suri) [reply services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:40]: done

File: services/video-conferencing-service/src/enums/index copy.ts (hunk 9 above)
Comment (vineet-suri) [reply services/video-conferencing-service/src/enums/index copy.ts:1]: done

File: services/video-conferencing-service/src/datasources/videochat-db.datasource.ts (hunk 10 above)
Comment (vineet-suri) [reply services/video-conferencing-service/src/datasources/videochat-db.datasource.ts:6]: done

File: services/video-conferencing-service/package.json [hunk 12]
```diff
@@ -24,39 +25,64 @@
...
     "dist",
     "src",
     "!*/__tests__"
   ],
   "dependencies": {
-    "@loopback/boot": "^2.3.1",
-    "@loopback/context": "^3.8.1",
-    "@loopback/core": "^2.7.0",
-    "@loopback/repository": "^2.5.1",
+    "@loopback/boot": "^2.0.0",
```
Comment (samarpan-b) [inline services/video-conferencing-service/package.json:51]: why downgrade ?

File: services/video-conferencing-service/package.json [hunk 13]
```diff
@@ -24,39 +25,64 @@
     "pretest": "npm run clean && npm run build",
     "test": "lb-mocha --allow-console-logs \"dist/__tests__\"",
     "posttest": "npm run lint",
-    "test:dev": "lb-mocha --allow-console-logs dist/__tests__/**/*.js && npm run posttest",
-    "clean": "lb-clean dist *.tsbuildinfo .eslintcache"
+    "test:dev": "lb-mocha --allow-console-logs dist/__tests__/**/*.js && npm run posttest",    
+    "migrate": "node ./dist/migrate",
+    "prestart": "npm run build",
+    "start": "node -r source-map-support/register .",
```
Comment (samarpan-b) [inline services/video-conferencing-service/package.json:31]: this is a component. It cannot have start

File: services/video-conferencing-service/src/controllers/video-chat-archive.controller.ts [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+            },
+          },
+        },
+      },
+    },
+  })
+  async deleteArchive(
+    @param.path.string('archiveId') archiveId: string,
+  ): Promise<void> {
+    await this.videoChatProvider.getArchives(archiveId);
```
Comment (samarpan-b) [inline services/video-conferencing-service/src/controllers/video-chat-archive.controller.ts:72]: why get ?

File: services/video-conferencing-service/src/enums/content-type.enum.ts [hunk 15]
```diff
@@ -0,0 +1,4 @@
+export const enum CONTENT_TYPE {
```
Comment (samarpan-b) [inline services/video-conferencing-service/src/enums/content-type.enum.ts:1]: there is already one enum like this in core. use that

File: services/video-conferencing-service/src/enums/status-codes.enum.ts [hunk 16]
```diff
@@ -0,0 +1,28 @@
+export const enum STATUS_CODE {
```
Comment (samarpan-b) [inline services/video-conferencing-service/src/enums/status-codes.enum.ts:1]: use the same from core

File: services/video-conferencing-service/src/models/video-chat-session.model.ts [hunk 17]
```diff
@@ -0,0 +1,78 @@
+import {model, property} from '@loopback/repository';
+import {UserModifiableEntity} from '@sourcefuse-service-catalog/core';
+
+@model({
+  name: 'video-session-details',
```
Comment (samarpan-b) [inline services/video-conferencing-service/src/models/video-chat-session.model.ts:5]: table names should be in snake case

File: services/video-conferencing-service/src/providers/bearer/index.ts [hunk 18]
```diff
@@ -0,0 +1,13 @@
+import {Provider} from '@loopback/context';
+import {verify} from 'jsonwebtoken';
+import {VerifyFunction} from 'loopback4-authentication';
+import {IAuthUserWithPermissions} from '@sourcefuse-service-catalog/core';
+export class BearerTokenVerifyProvider
```
Comment (samarpan-b) [inline services/video-conferencing-service/src/providers/bearer/index.ts:5]: No need. use from core

File: services/video-conferencing-service/src/providers/vonage/vonage.provider.ts [hunk 19]
```diff
@@ -0,0 +1,58 @@
+import {Provider} from '@loopback/core';
+import {HttpErrors} from '@loopback/rest';
+import {VonageEnums} from '../../enums/video-chat.enum';
+import {MeetingOptions, SessionOptions} from '../../types';
+import {VonageVideoChat} from './types';
+import OpenTok = require('opentok'); // https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require
```
Comment (samarpan-b) [inline services/video-conferencing-service/src/providers/vonage/vonage.provider.ts:6]: why this ?

File: services/video-conferencing-service/tsconfig.json [hunk 20]
```diff
@@ -2,6 +2,8 @@
   "$schema": "http://json.schemastore.org/tsconfig",
   "extends": "@loopback/build/config/tsconfig.common.json",
   "compilerOptions": {
+    "experimentalDecorators": true,
```
Comment (samarpan-b) [inline services/video-conferencing-service/tsconfig.json:5]: no need. They are already available in base json @loopback/build/config/tsconfig.common.json

**Comment by samarpan-b** [comment]
Fix sonar issues

File: services/video-conferencing-service/src/controllers/video-chat-archive.controller.ts (hunk 14 above)
Comment (bhavyasf) [reply services/video-conferencing-service/src/controllers/video-chat-archive.controller.ts:72]: @samarpan-b typo error. It is delete in SFO-51 branch!

**Pull request merged**